src/GDGUkraine/utils.py

```diff
@@ -114,3 +122,283 @@ def make_vcard(user_reg, url=None):
             url=url))
+
+
+def uri_builder(rparams, *args, **kwargs):
+    '''
+    *args and **kwargs are checked for integrity with corresponding handler
+    '''
+
+    params = rparams['args'].copy()
+    url = rparams['url']
+
+    ikwargs = kwargs.copy()
+    iargs = list(args)
+
+    rkwargs = {}
+    rargs = []
+
+    # Match url_for's input params to real handler's signature and put them
+    # into variables holding separate URI path parts along with
+    # GET key-value params
+    while len(params):
+        param = params.popitem(last=False)[1]
+        if param.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD:
+            # param goes prior to *args
+            if param.name in ikwargs:
+                rargs.append(ikwargs.pop(param.name))
+            elif len(iargs):
+                rargs.append(iargs.pop(0))
+            elif param.default is inspect.Parameter.empty:
+                raise TypeError
+            else:
+                rargs.append(param.default)
+        elif param.kind == inspect.Parameter.VAR_POSITIONAL:
+            # param is *args
+            rargs.extend(iargs)
+            iargs.clear()  # or maybe it's better to del it?
+        elif param.kind == inspect.Parameter.KEYWORD_ONLY:
+            # param is between (* or *args) and **kwargs
+            if param.name in rkwargs:
+                raise TypeError(
+                    'Got multiple values for argument `{}`'.format(param.name))
+            elif param.name not in ikwargs:
+                if param.default is inspect.Parameter.empty:
+                    raise TypeError(
+                        'Missing required argument `{}`'.format(param.name))
+                else:
+                    rkwargs[param.name] = param.default
+            else:
+                rkwargs[param.name] = ikwargs.pop(param.name)
+        elif param.kind == inspect.Parameter.VAR_KEYWORD:
+            # param is **kwargs
+            rkwargs.update(ikwargs)
+            ikwargs.clear()
+
+    # Check whether there's any params exceeding ones
+    # declared into original handler
+    if len(iargs):
+        raise TypeError('Too many positional arguments passed!')
+    elif len(ikwargs):
+        raise TypeError('Too many keyword arguments passed!')
+
+    # Build URI path string ending for concatenation with base URI
+    uargs = '/'.join([urllib.parse.quote_plus(_)
+                      for _ in rargs if _])
+    # Build GET params list string
+    ukwargs = '&'.join(['='.join([urllib.parse.quote_plus(k),
+                                  urllib.parse.quote_plus(str(v))])
+                        for k, v in rkwargs.items() if v])
+
+    if uargs:
+        url = '/'.join([url, uargs])
+
+    if ukwargs:
+        url = '?'.join([url, ukwargs])
+
+    # Return final URI string
+    return url
+
+
+def build_url_map(force=False):
+    '''Builds resolve map for class-based routes
+        build_url_map(force=True) is called by url map builder cherrypy plugin
```

Comment:
This line should be in that plugin comment with the explanation why `build_url_map` is called with `force=True`, not here.